tensorflow_datasets/core/load.py

```diff
@@ -382,7 +382,7 @@ def find_builder_dir(
 
   # If version not given, extract the version
   if not version_str:
```

Comment:
`if not version_str` means that _get_version_str will never be called with `version_str is not None`. Haven't the tests catched this error ?

Reply:
Added some tests.
It failed only was `version_str` was a regex.